store/driver/txn/txn_driver.go

```diff
@@ -75,25 +76,57 @@ func (txn *tikvTxn) Commit(ctx context.Context) error {
 
 // GetSnapshot returns the Snapshot binding to this transaction.
 func (txn *tikvTxn) GetSnapshot() kv.Snapshot {
-	return &tikvSnapshot{txn.KVTxn.GetSnapshot(), nil}
+	return &tikvSnapshot{txn.KVTxn.GetSnapshot(), txn.snapshotInterceptor}
 }
 
 // Iter creates an Iterator positioned on the first entry that k <= entry's key.
 // If such entry is not found, it returns an invalid Iterator with no error.
 // It yields only keys that < upperBound. If upperBound is nil, it means the upperBound is unbounded.
 // The Iterator must be Closed after use.
-func (txn *tikvTxn) Iter(k kv.Key, upperBound kv.Key) (kv.Iterator, error) {
-	it, err := txn.KVTxn.Iter(k, upperBound)
-	return newKVIterator(it), derr.ToTiDBErr(err)
+func (txn *tikvTxn) Iter(k kv.Key, upperBound kv.Key) (iter kv.Iterator, err error) {
+	var dirtyIter, snapIter kv.Iterator
+
+	if dirtyIter, err = txn.GetMemBuffer().Iter(k, upperBound); err != nil {
```

Comment:
If `dirtyIter` is nil, we can use `snapIter` directly for the sake of performance?